src/header/Header.tsx

```diff
@@ -30,98 +33,62 @@ export function Header() {
+        <StandardListItem aria-label={name} data-key={index}>
+          {name}
+        </StandardListItem>
+      );
+    }),
+    <StandardListItem aria-label="all-clusters">
+      {t('clusters.overview.title-all-clusters')}
+    </StandardListItem>,
   ];
 
-  const getNamespaceLabel = () => {
-    if (activeNamespace === '-all-') return t('navigation.all-namespaces');
-    else return activeNamespace || t('navigation.select-namespace');
-  };
-
-  let headerActions = [];
-  if (cluster) {
-    headerActions.push({
-      glyph: 'megamenu',
-      label: getNamespaceLabel(),
-      notificationCount: 0,
-      callback: () => {
-        refetch();
-        setIsNamespaceOpen(!isNamespaceOpen);
-      },
-      menu: (
-        <NamespaceDropdown hideDropdown={() => setIsNamespaceOpen(false)} />
-      ),
-    });
-  }
-
-  if (isFeedbackEnabled) {
-    headerActions.unshift({
-      glyph: 'feedback',
-      notificationCount: 0,
-      callback: () => {
-        window.open(feedbackLink, '_blank');
-      },
-    });
-  }
-
   return (
-    <Shellbar
-      className="header"
-      logo={
-        <>
-          <SidebarSwitcher />
-          <Logo />
-        </>
-      }
-      productTitle={cluster?.contextName || cluster?.name}
-      productMenu={clustersList}
-      profile={{
-        glyph: 'customer',
-        colorAccent: 10,
-      }}
-      actions={headerActions}
-      profileMenu={[
-        {
-          name: t('navigation.preferences.title'),
-          callback: () => setPreferencesOpen(true),
-        },
-      ]}
-      // @ts-ignore
-      popoverPropsFor={{
-        profileMenu: { 'aria-label': 'topnav-profile-btn' },
-        actionMenu: {
-          show: isNamespaceOpen,
-          onClickOutside: () => setIsNamespaceOpen(false),
-          onEscapeKey: () => setIsNamespaceOpen(false),
-        },
-        productMenu: {
-          show: isClustersOpen,
-          onClickOutside: () => setIsClustersOpen(false),
-          onEscapeKey: () => setIsClustersOpen(false),
-          onClick: () => setIsClustersOpen(!isClustersOpen),
-        },
-      }}
-    />
+    <>
+      <ShellBar
+        className="header"
+        startButton={<SidebarSwitcher />}
+        onLogoClick={() => navigate('/clusters')}
+        logo={<Logo />}
+        primaryTitle={cluster?.contextName || cluster?.name}
+        menuItems={clustersList}
+        onMenuItemClick={e =>
+          e.detail.item.textContent ===
+          t('clusters.overview.title-all-clusters')
+            ? navigate('/clusters')
+            : navigate(`/cluster/${e.detail.item.textContent}`)
+        }
+        profile={
+          <Avatar
+            icon="customer"
+            colorScheme="Accent10"
+            accessibleName="Preferences"
+          />
+        }
+        onProfileClick={() => setPreferencesOpen(true)}
+      >
+        <ShellBarItem
+          onClick={() => setShowAdd(true)}
+          icon="add"
+          text="Upload YAML"
+        />
```

Comment:
Disable or hide this button when in the cluster overview?
same thing then also for the sidebarswitcher, but more important for Add-Button